[BUG] is ENABLE_HAICRYPT_LOGGING really supported?

**Describe the bug**
When I enable ENABLE_HAICRYPT_LOGGING compile option in cmake, it is impossible to build the code because HaiCrypt_DumpConfig (written in 2017) refers to HaiCrypt_Cfg.cipher field, but the structure HaiCrypt_Cfg (written in 2018) doesn't have this field

**To Reproduce**

```
$ git log | head -4
commit 277fa583c97c614785710980f3ad0c036a960c29
Author: Maxim Sharabayko <maxlovic@gmail.com>
Date:   Wed Mar 30 16:52:48 2022 +0200
```

```
$ cmake . -DENABLE_HEAVY_LOGGING=1 -DENABLE_HAICRYPT_LOGGING=2 -DENABLE_LOGGING=1
-- The C compiler identification is AppleClang 13.0.0.13000029
-- The CXX compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
...
-- C++ STD: Forcing C++11 on applications
-- DETECTED SYSTEM: DARWIN
CMake Warning at CMakeLists.txt:665 (message):
   *** ENABLED INSECURE HAICRYPT LOGGING - USE FOR TESTING ONLY!!! ***
...
-- Generating done
-- Build files have been written to: /Users/max/srt
```

```
$ make
[  1%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/api.cpp.o
[  3%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/buffer.cpp.o
...
[ 65%] Building C object CMakeFiles/srt_virtual.dir/haicrypt/hcrypt_xpt_srt.c.o
[ 67%] Building CXX object CMakeFiles/srt_virtual.dir/haicrypt/haicrypt_log.cpp.o
/Users/max/Sites/srt/haicrypt/haicrypt_log.cpp:100:22: error: no member named 'cipher' in 'HaiCrypt_Cfg'
            << (cfg->cipher == HaiCryptCipher_OpenSSL_EVP_CTR() ? "OSSL-EVP-CTR":
                ~~~  ^
/Users/max/Sites/srt/haicrypt/../srtcore/logging.h:58:72: note: expanded from macro 'LOGC'
    const srt_logging::LogDispatcher::Proxy& log_prox SRT_ATR_UNUSED = args; \
                                                                       ^~~~
/Users/max/Sites/srt/haicrypt/haicrypt_log.cpp:100:32: error: use of undeclared identifier 'HaiCryptCipher_OpenSSL_EVP_CTR'
            << (cfg->cipher == HaiCryptCipher_OpenSSL_EVP_CTR() ? "OSSL-EVP-CTR":
                               ^
/Users/max/Sites/srt/haicrypt/haicrypt_log.cpp:101:22: error: no member named 'cipher' in 'HaiCrypt_Cfg'
                cfg->cipher == HaiCryptCipher_OpenSSL_AES() ? "OSSL-AES":
                ~~~  ^
/Users/max/Sites/srt/haicrypt/../srtcore/logging.h:58:72: note: expanded from macro 'LOGC'
    const srt_logging::LogDispatcher::Proxy& log_prox SRT_ATR_UNUSED = args; \
                                                                       ^~~~
/Users/max/Sites/srt/haicrypt/haicrypt_log.cpp:101:32: error: use of undeclared identifier 'HaiCryptCipher_OpenSSL_AES'
                cfg->cipher == HaiCryptCipher_OpenSSL_AES() ? "OSSL-AES":
                               ^
/Users/max/Sites/srt/haicrypt/haicrypt_log.cpp:104:22: error: no member named 'cipher' in 'HaiCrypt_Cfg'
                cfg->cipher == HaiCryptCipher_Get_Instance() ? "Nettle-AES":
                ~~~  ^
/Users/max/Sites/srt/haicrypt/../srtcore/logging.h:58:72: note: expanded from macro 'LOGC'
    const srt_logging::LogDispatcher::Proxy& log_prox SRT_ATR_UNUSED = args; \
                                                                       ^~~~
/Users/max/Sites/srt/haicrypt/haicrypt_log.cpp:104:32: error: use of undeclared identifier 'HaiCryptCipher_Get_Instance'; did you mean 'HaiCryptCryspr_Get_Instance'?
                cfg->cipher == HaiCryptCipher_Get_Instance() ? "Nettle-AES":
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                               HaiCryptCryspr_Get_Instance
/Users/max/Sites/srt/haicrypt/../srtcore/logging.h:58:72: note: expanded from macro 'LOGC'
    const srt_logging::LogDispatcher::Proxy& log_prox SRT_ATR_UNUSED = args; \
                                                                       ^
/Users/max/Sites/srt/haicrypt/haicrypt.h:32:17: note: 'HaiCryptCryspr_Get_Instance' declared here
HaiCrypt_Cryspr HaiCryptCryspr_Get_Instance (void);     /* Return a default cryspr instance */
                ^
6 errors generated.
make[2]: *** [CMakeFiles/srt_virtual.dir/haicrypt/haicrypt_log.cpp.o] Error 1
make[1]: *** [CMakeFiles/srt_virtual.dir/all] Error 2
make: *** [all] Error 2
```


**Expected behavior**

I expect that srt will compile and debug info about generated keys.

--- a/haicrypt/cryspr-config.h
+++ b/haicrypt/cryspr-config.h
@@ -12,7 +12,7 @@
 #elif defined(USE_GNUTLS)
 #include "cryspr-gnutls.h"
 #define cryspr4SRT()  crysprGnuTLS()
-#define CRYSPR_IMPL_DESC "GbuTLS"
+#define CRYSPR_IMPL_DESC "GnuTLS"
 #elif defined(USE_MBEDTLS)
 #include "cryspr-mbedtls.h"
 #define cryspr4SRT() crysprMbedtls()

--- a/haicrypt/cryspr-config.h
+++ b/haicrypt/cryspr-config.h
@@ -8,14 +8,18 @@
 #if defined(USE_OPENSSL)
 #include "cryspr-openssl.h"
 #define cryspr4SRT()  crysprOpenSSL()
+#define CRYSPR_IMPL_DESC "OpenSSL-AES"
 #elif defined(USE_GNUTLS)
 #include "cryspr-gnutls.h"
 #define cryspr4SRT()  crysprGnuTLS()
+#define CRYSPR_IMPL_DESC "GbuTLS"
 #elif defined(USE_MBEDTLS)
 #include "cryspr-mbedtls.h"
 #define cryspr4SRT() crysprMbedtls()
+#define CRYSPR_IMPL_DESC "MbedTLS"
 #else
 #error Cryspr implementation not selected. Please define USE_* + OPENSSL/GNUTLS/MBEDTLS.
+#define CRYSPR_IMPL_DESC "No Cipher"
 #endif
 
 

--- a/haicrypt/haicrypt_log.cpp
+++ b/haicrypt/haicrypt_log.cpp
@@ -99,12 +99,7 @@ void HaiCrypt_DumpConfig(const HaiCrypt_Cfg* cfg)
     LOGC(hclog.Debug, log << "CFG DUMP: flags=" << cfg_flags.str()
             << " xport=" << (cfg->xport == HAICRYPT_XPT_SRT ? "SRT" : "INVALID")
             << " cipher="
-            << (cfg->cipher == HaiCryptCipher_OpenSSL_EVP_CTR() ? "OSSL-EVP-CTR":
-                cfg->cipher == HaiCryptCipher_OpenSSL_AES() ? "OSSL-AES":
-                // This below is used as the only one when Nettle is used. When OpenSSL
-                // is used, one of the above will trigger, and the one below will then never trigger.
-                cfg->cipher == HaiCryptCipher_Get_Instance() ? "Nettle-AES":
-                "UNKNOWN")
+            << CRYSPR_IMPL_DESC
             << " key_len=" << cfg->key_len << " data_max_len=" << cfg->data_max_len);
 
 

--- a/srtcore/logging.h
+++ b/srtcore/logging.h
@@ -247,6 +247,11 @@ public:
             return *this;
         }
 
+        DummyProxy& vform(const char*, va_list)
+        {
+            return *this;
+        }
+
         DummyProxy& setloc(const char* , int , std::string)
         {
             return *this;

